app/src/main/java/com/mstudio/superheroarch/MainActivity.kt

```diff
@@ -43,7 +38,6 @@ class MainActivity : AppCompatActivity() {
         }
 
         val characterRecyclerView = findViewById<RecyclerView>(R.id.characters_recycler)
-
         characterRecyclerView.adapter = adapter
 
         adapter.setOnItemClickListener(object : CharacterAdapter.OnItemClickListener {
```

Comment:
the viewModel should be notified about the clicks, even if those clicks are in an adapter